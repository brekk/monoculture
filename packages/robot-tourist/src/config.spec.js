import {
  LOGO,
  CONFIG,
  CERTAIN_COMMON_WORDS,
  USER_DEFINED_VALUES,
  DEFAULT_CONFIG,
  HELP_CONFIG,
} from './config'

test('LOGO', () => {
  expect(LOGO).toMatchSnapshot()
})
test('CONFIG', () => {
  expect(CONFIG).toMatchSnapshot()
})
test('CERTAIN_COMMON_WORDS', () => {
  expect(CERTAIN_COMMON_WORDS).toMatchSnapshot()
})
test('USER_DEFINED_VALUES', () => {
  expect(USER_DEFINED_VALUES).toMatchSnapshot()
})
test('DEFAULT_CONFIG', () => {
  expect(DEFAULT_CONFIG).toMatchSnapshot()
})
test('HELP_CONFIG', () => {
  expect(HELP_CONFIG).toMatchSnapshot()
})
