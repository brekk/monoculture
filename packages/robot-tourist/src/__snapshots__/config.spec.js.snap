// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CERTAIN_COMMON_WORDS 1`] = `
[
  "use",
  "get",
  "id",
]
`;

exports[`CONFIG 1`] = `
{
  "alias": {
    "assumeSimilarWords": [
      "similar",
      "a",
    ],
    "dropJSKeywords": [
      "j",
    ],
    "dropStrings": [
      "s",
    ],
    "dropTSKeywords": [
      "J",
    ],
    "fun": [
      "f",
    ],
    "help": [
      "h",
    ],
    "histogramMinimum": [
      "hits",
      "m",
    ],
    "ignore": [
      "i",
    ],
    "ignoreTokens": [
      "t",
    ],
    "limit": [
      "l",
    ],
    "skipWords": [
      "w",
    ],
  },
  "array": [
    "skipWords",
    "ignoreTokens",
    "ignore",
  ],
  "boolean": [
    "dropStrings",
    "assumeSimilarWords",
    "dropJSKeywords",
    "dropTSKeywords",
    "fun",
  ],
  "configuration": {
    "strip-aliased": true,
  },
  "number": [
    "histogramMinimum",
  ],
}
`;

exports[`DEFAULT_CONFIG 1`] = `
{
  "assumeSimilarWords": true,
  "dropImports": true,
  "dropJSKeywords": true,
  "dropStrings": true,
  "dropTSKeywords": true,
  "fun": true,
  "help": false,
  "histogramMinimum": 1,
  "ignore": [],
  "ignoreTokens": [
    "use",
    "get",
    "id",
  ],
  "limit": Infinity,
  "skipWords": [],
}
`;

exports[`HELP_CONFIG 1`] = `
{
  "assumeSimilarWords": "Should we attempt to infer whether words are similar?",
  "dropJSKeywords": "Should JS keywords be dropped? (default: true)",
  "dropStrings": "Should strings be retained or removed?",
  "dropTSKeywords": "Should TS keywords be dropped? (default: true)",
  "fun": "Show the robot",
  "help": "This text!",
  "histogramMinimum": "What is the minimum number of matches as a threshold?",
  "ignore": "Ignore a matched string (can be specified multiple times)",
  "ignoreTokens": "Ignore a token once it is processed (more reliable than \`--ignore\`)",
  "limit": "What top number of words do you want to see? (default: Infinity)",
  "skipWords": "Ignore a given word (this should be merged with --ignore)",
}
`;

exports[`LOGO 1`] = `
"   /\\/\\
 .======.
<dIIIIIIb>
 |[o><o]|
 (||||||)
 |______|
"
`;

exports[`USER_DEFINED_VALUES 1`] = `[]`;
